src/differential_value_iteration/algorithms/mdvi.py

```diff
@@ -0,0 +1,69 @@
+"""Evaluation and Control implementations of M? Differential Value Iteration."""
+from typing import Union
+
+import numpy as np
+from absl import logging
+from differential_value_iteration.algorithms import algorithm
+from differential_value_iteration.environments import structure
+
+
+class Evaluation(algorithm.Evaluation):
+  def __init__(
+      self,
+      mrp: structure.MarkovRewardProcess,
+      initial_values: np.ndarray,
+      initial_r_bar: Union[float, np.ndarray],
+      step_size: float,
+      beta: float,
+      synchronized: bool):
+    self.mrp = mrp
+    self.initial_values = initial_values.copy()
+    if isinstance(initial_r_bar, np.ndarray):
+      self.initial_r_bar = initial_r_bar.copy()
+    else:
+      self.initial_r_bar = np.full(shape=mrp.num_states,
+                                   fill_value=initial_r_bar, dtype=np.float32)
+    self.step_size = step_size
+    self.beta = beta
+    self.index = 0
+    self.synchronized = synchronized
+    self.current_values = None
+    self.r_bar = None
+    self.reset()
+
+  def reset(self):
+    self.current_values = self.initial_values.copy()
+    self.r_bar = self.initial_r_bar.copy()
+
+  def diverged(self) -> bool:
```

Comment:
I think this function does an important job of differentiating between 'not converged yet' and 'diverged'!